internal/testrunner/runners/system/servicedeployer/custom_agent.go

```diff
@@ -4,33 +4,245 @@
 
 package servicedeployer
 
+import (
+	_ "embed"
+	"fmt"
+	"os"
+
+	"github.com/elastic/elastic-package/internal/compose"
+	"github.com/elastic/elastic-package/internal/docker"
+	"github.com/elastic/elastic-package/internal/files"
+	"github.com/elastic/elastic-package/internal/install"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/stack"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed custom-agent-base-config.yml
+var customAgentYml string
+
 // CustomAgentDeployer knows how to deploy a custom elastic-agent defined via
 // a Docker Compose file.
 type CustomAgentDeployer struct {
-	cd *DockerComposeServiceDeployer
+	cfg      string
+	ymlPaths []string
+}
```

Comment:
Umm, in the previous version of the code this `CustomAgentDeployer` was a wrapper over `DockerComposeServiceDeployer`.
Are there more differences apart of the additional config?

Reply:
Also is needed to send appconfig env vars, but I think it makes sense to try to use a wrapper approach if possible. Will work on that.